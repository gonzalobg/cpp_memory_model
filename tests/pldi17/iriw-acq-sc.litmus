C iriw-acq-sc
{ [x] = 0; [y] = 0;  }

P0 (atomic_int* x) {
  atomic_store_explicit(x, 1, memory_order_seq_cst);
}

P1 (atomic_int* x, atomic_int* y) {
  int a = atomic_load_explicit(x, memory_order_acquire);
  int c = atomic_load_explicit(y, memory_order_seq_cst);
}

P2 (atomic_int* x, atomic_int* y) {
  int b = atomic_load_explicit(y, memory_order_acquire);
  int d = atomic_load_explicit(x, memory_order_seq_cst);
}

P3 (atomic_int* y) {
  atomic_store_explicit(y, 1, memory_order_seq_cst);
}

exists(1:a=1 /\ 2:b=1 /\ 1:c=0 /\ 2:d=0)