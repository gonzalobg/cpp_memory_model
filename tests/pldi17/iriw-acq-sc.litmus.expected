Test iriw-acq-sc Allowed
States 16
1:a=0; 1:c=0; 2:b=0; 2:d=0;
1:a=0; 1:c=0; 2:b=0; 2:d=1;
1:a=0; 1:c=0; 2:b=1; 2:d=0;
1:a=0; 1:c=0; 2:b=1; 2:d=1;
1:a=0; 1:c=1; 2:b=0; 2:d=0;
1:a=0; 1:c=1; 2:b=0; 2:d=1;
1:a=0; 1:c=1; 2:b=1; 2:d=0;
1:a=0; 1:c=1; 2:b=1; 2:d=1;
1:a=1; 1:c=0; 2:b=0; 2:d=0;
1:a=1; 1:c=0; 2:b=0; 2:d=1;
1:a=1; 1:c=0; 2:b=1; 2:d=0;
1:a=1; 1:c=0; 2:b=1; 2:d=1;
1:a=1; 1:c=1; 2:b=0; 2:d=0;
1:a=1; 1:c=1; 2:b=0; 2:d=1;
1:a=1; 1:c=1; 2:b=1; 2:d=0;
1:a=1; 1:c=1; 2:b=1; 2:d=1;
Ok
Witnesses
Positive: 1 Negative: 15
Condition exists (1:a=1 /\ 2:b=1 /\ 1:c=0 /\ 2:d=0)
Observation iriw-acq-sc Sometimes 1 15
Hash=092ac43db26f7a4f2fa03fd9a05d5951

